src/integrationTest/java/uk/gov/hmcts/darts/task/service/InboundToUnstructuredProcessorIntTest.java

```diff
@@ -0,0 +1,177 @@
+    private ExternalObjectDirectoryStub externalObjectDirectoryStub;
+
+    @SpyBean
+    ExternalObjectDirectoryRepository eodRepository;
+    @MockBean
+    DataManagementService dataManagementService;
+
+    @Autowired
+    private InboundToUnstructuredProcessor inboundToUnstructuredProcessor;
+
+    @BeforeEach
+    public void setup() {
+        externalObjectDirectoryStub = dartsDatabase.getExternalObjectDirectoryStub();
+        when(dataManagementService.getBlobData(any(), any())).thenReturn(getBinaryTranscriptionDocumentData());
+        when(dataManagementService.saveBlobData(any(), any())).thenReturn(UUID.randomUUID());
+    }
+
+    @Test
+    void processInboundMediasToUnstructured() {
+        // given
+        List<MediaEntity> medias = dartsDatabase.getMediaStub().createAndSaveSomeMedias();
+
+        //matches because no corresponding unstructured
+        var media1 = medias.get(0);
+        externalObjectDirectoryStub.createAndSaveEod(media1, STORED, INBOUND);
+
+        //matches because unstructured failed with no max attempts reached
+        var media2 = medias.get(1);
+        externalObjectDirectoryStub.createAndSaveEod(media2, STORED, INBOUND);
+        externalObjectDirectoryStub.createAndSaveEod(media2, FAILURE, UNSTRUCTURED);
+
+        //does not match because corresponding unstructured is stored
+        var media3 = medias.get(2);
+        externalObjectDirectoryStub.createAndSaveEod(media3, STORED, INBOUND);
+        externalObjectDirectoryStub.createAndSaveEod(media3, STORED, UNSTRUCTURED);
+
+        //does not match because unstructured failed with max attempts reached
+        var media4 = medias.get(3);
+        externalObjectDirectoryStub.createAndSaveEod(media4, STORED, INBOUND);
+        var failed = externalObjectDirectoryStub.createAndSaveEod(media4, FAILURE, UNSTRUCTURED);
+        failed.setTransferAttempts(10);
+        eodRepository.save(failed);
+
+        // when
+        inboundToUnstructuredProcessor.processInboundToUnstructured();
+
+        // then
+        assertThat(externalObjectDirectoryStub.findByMediaStatusAndType(media1, STORED, UNSTRUCTURED)).hasSize(1);
+        assertThat(externalObjectDirectoryStub.findByMediaStatusAndType(media2, STORED, UNSTRUCTURED)).hasSize(1);
+        var argument = ArgumentCaptor.forClass(ExternalObjectDirectoryEntity.class);
+        verify(eodRepository, atLeastOnce()).saveAndFlush(argument.capture());
+        List<ExternalObjectDirectoryEntity> createdUnstructured = argument.getAllValues();
+        List<Integer> createdUnstructuredMediaIds = createdUnstructured.stream().map(eod -> eod.getMedia().getId()).collect(toList());
+        assertThat(createdUnstructuredMediaIds).contains(media1.getId(), media2.getId());
+        assertThat(createdUnstructuredMediaIds).doesNotContain(media3.getId());
+        assertThat(createdUnstructuredMediaIds).doesNotContain(media4.getId());
+    }
+
+    @Test
+    void processInboundTranscriptionDocumentToUnstructured() {
```

Comment:
Author note:
I started adding equivalent integration tests for Annotation entity and Case document entity, but it's quite a tedious process since we don't have test fixtures methods already available that I can use. 
If the goal is to fix the bug asap, what we have will do the job since transcription documents, annotation documents   and case documents are all treated the same way.